discovery-infra/tests/base_test.py

```diff
@@ -12,31 +12,39 @@
 from test_infra import consts, utils
 from test_infra.helper_classes.cluster import Cluster
 
-from tests.conftest import env_variables
-
 
 def random_name():
     return ''.join(random.choice(ascii_lowercase) for i in range(10))
 
 
 class BaseTest:
-    @pytest.fixture(scope="function")
-    def node_controller(self, setup_node_controller):
-        controller = setup_node_controller
-        controller.set_correct_boot_order_to_all_nodes()
-        yield controller
-        controller.shutdown_all_nodes()
-        controller.format_all_node_disks()
+    @pytest.fixture()
+    def node_controller(self, node_controller_class):
```

Comment:
we deleted nodecontroller fixture, not sure i agree with this one.
It looks much more complicated and for no reason. I like session scope fixture more